### PR TITLE
Add YANG model for FABRIC_PORT

### PR DESCRIPTION
**Pull request by kenneth-arista**
#### Why I did it

Introduce YANG model for FABRIC in CONFIG_DB, which was added in https://github.com/sonic-net/sonic-buildimage/pull/14170.

#### How I did it

This is a clone of @jfeng-arista's PR https://github.com/sonic-net/sonic-buildimage/pull/14282 to resolve conflicts with upstream changes. 

#### How to verify it

Passing pipeline build is sufficient.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

### CODE REVIEW

**Comment by kenneth-arista** [comment]
@gechiang

**Comment by kenneth-arista** [comment]
@arlakshm @jfeng-arista

**Review by arlakshm — DISMISSED** [review]

File: src/sonic-yang-models/yang-models/sonic-fabric-port.yang [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+                leaf name {
+                    type string {
+                        length 1..128;
+                    }
+                }
+
+                leaf isolateStatus {
+                    description "Isolation status of a fabric port";
+                    type string {
```
Comment (ganglyu) [inline src/sonic-yang-models/yang-models/sonic-fabric-port.yang:32]: Why not use Boolean?

Comment (kenneth-arista) [reply src/sonic-yang-models/yang-models/sonic-fabric-port.yang:32]: Good suggestion. I will revise it to use `type stypes:boolean_type` where stypes is sonic-types.

Comment (kenneth-arista) [reply src/sonic-yang-models/yang-models/sonic-fabric-port.yang]: Done. See latest commit. I'm waiting for the pipeline build to complete to check if any further revisions are needed.

**Review by arlakshm — APPROVED** [review]

**Review by gechiang — APPROVED** [review]

**Review by qiluo-msft — APPROVED** [review]

**Comment by gechiang** [comment]
@kenneth-arista , @jfeng-arista , This PR also can not be clean cherry picked tot he azure/sonic-buildimage-msft repo 202205 branch.
Can you please raise a PR in the azure/sonic-buildimage-msft repo directly and tag me to review/merge.
Thanks!

**Comment by jfeng-arista** [comment]
will do thank you

Jie

**Comment by StormLiangMS** [comment]
@gechiang  do we need this to cherry pick to 202211/202305? could we have ADO to track?

**Comment by gechiang** [comment]
@StormLiangMS 
MSFT ADO: 24836849

**Comment by gechiang** [comment]
Removed request for 202211 due to this is treated as "feature" and not bug fix back port.  Same reason as why it was not picked for 202205.

**Pull request merged**